FormalConjectures/ErdosProblems/340.lean

```diff
@@ -0,0 +1,175 @@
+example : greedySidon 10 = 97 := by
+  decide +native
+
+/--
+Let $A = \{1, 2, 4, 8, 13, 21, 31, 45, 66, 81, 97, \ldots\}$ be the greedy Sidon sequence:
+we begin with $1$ and iteratively include the next smallest integer that preserves the
+Sidon property (i.e. there are no non-trivial solutions to $a + b = c + d$). What is the
+order of growth of $A$? Is it true that $| A \cap\{1, \ldots, N\}| \gg N^{1/2−\varepsilon}$
+for all $\varepsilon > 0$ and large $N$?
+-/
+@[category research open, AMS 5]
+theorem erdos_340 (ε : ℝ) (hε : ε > 0) :
+    (fun n : ℕ ↦ √n / n ^ ε) =O[atTop]
+      fun n : ℕ ↦ ((Set.range greedySidon ∩ Set.Icc 1 n).ncard : ℝ) := by
+  sorry
+
+/--
+Let $A = \{1, 2, 4, 8, 13, 21, 31, 45, 66, 81, 97, \ldots\}$ be the greedy Sidon sequence:
+we begin with $1$ and iteratively include the next smallest integer that preserves the
+Sidon property (i.e. there are no non-trivial solutions to $a + b = c + d$). What is the
+order of growth of $A$? Is it true that $| A \cap\{1, \ldots, N\}| \gg N^{1/2−\varepsilon}$
+for all $\varepsilon > 0$ and large $N$?
+-/
+@[category research open, AMS 5]
+theorem erdos_340.variants.isTheta (ε : ℝ) (hε : ε > 0) :
+    (fun n : ℕ ↦ ((Set.range greedySidon ∩ Set.Icc 1 n).ncard : ℝ)) =Θ[atTop]
+      (answer(sorry) : ℕ → ℝ) := by
+  sorry
+
+/--
+It is trivial that this sequence grows at least like $\gg N^{1/3}$.
+-/
+@[category undergraduate, AMS 5]
+theorem erdos_340.variants.third (ε : ℝ) (hε : ε > 0) :
+    (fun n : ℕ ↦ (n : ℝ) ^ ((1 : ℝ) / 3)) =O[atTop]
+      fun n : ℕ ↦ ((Set.range greedySidon ∩ Set.Icc 1 n).ncard : ℝ) := by
+  sorry
+
+/--
+Erdős and Graham [ErGr80] also asked about the difference set $A - A$ and whether this has
+positive density.
+
+[ErGr80] Erdős, P. and Graham, R., Old and new problems and results in combinatorial number
+theory. Monographies de L'Enseignement Mathematique (1980).
+-/
+@[category research open, AMS 5]
+theorem erdos_340.variants.sub_hasPosDensity :
+    Set.HasPosDensity (Set.range greedySidon - Set.range greedySidon) :=
+  sorry
+
+/--
+Erdős and Graham [ErGr80] also asked about the difference set $A - A$ and whether this
+contains $22$, which it does.
+
+[ErGr80] Erdős, P. and Graham, R., Old and new problems and results in combinatorial number
+theory. Monographies de L'Enseignement Mathematique (1980).
+-/
+@[category research solved, AMS 5]
+theorem erdos_340.variants._22_mem_sub :
+    22 ∈ Set.range greedySidon - Set.range greedySidon := by
+  sorry
+
+/--
+The smallest integer which is unknown to be in $A - A$ is $33$.
+ -/
+@[category research open, AMS 5]
+theorem erdos_340.variants._33_mem_sub :
+    33 ∈ Set.range greedySidon - Set.range greedySidon ↔ answer(sorry) :=
+  sorry
+
+/--
+It may be true that all or almost all integers are in $A - A$.
+-/
+@[category research open, AMS 5]
+theorem erdos_340.variants.cofinite_sub :
+    (∀ᶠ n in cofinite, n ∈ Set.range greedySidon - Set.range greedySidon) ↔ answer(sorry) :=
+  sorry
+
```

Comment:
For these two, what do you think of adding a formalisation note indicating that there is some ambiguity in translating the "almost all" (and that there may be other reasonable non-equivalent formulations of this problem out there)